<launch>
    <arg name="fcu_url" default="/dev/ttyUSB0:1000000" />
    <arg name="gcs_url" default="" />
    <arg name="tgt_system" default="1" />
    <arg name="tgt_component" default="1" />
    <arg name="log_output" default="screen" />
    <arg name="fcu_protocol" default="v2.0" />
    <arg name="respawn_mavros" default="false" />
    <!-- 启动 PX4 -->
    <arg name="uav_id" default="XMM1"/>
    <arg name="simulation" default="false"/>
    <arg name="file_path" default="$(find test_controller)/json/myparams.json"/>
    <arg name="controller_rate" default="200"/>
    <arg name="observer_rate" default="200"/>
    <group unless="$(arg simulation)" ns="$(arg uav_id)">
        <include file="$(find mavros)/launch/px4.launch" >
            <arg name="fcu_url" value="$(arg fcu_url)" />
            <arg name="gcs_url" value="$(arg gcs_url)" />
            <arg name="tgt_system" value="$(arg tgt_system)" />
            <arg name="tgt_component" value="$(arg tgt_component)" />
            <arg name="log_output" value="$(arg log_output)" />
            <arg name="fcu_protocol" value="$(arg fcu_protocol)" />
            <arg name="respawn_mavros" value="$(arg respawn_mavros)" />
        </include>
    </group>
    <!-- 启动 Simu Python 脚本 -->
    <group if="$(arg simulation)">
        <node pkg="test_controller" type="simu.py" name="simu" output="screen">
        <param name="uav_id" value="$(arg uav_id)" />
    </node>
    </group>
    <!-- 启动 Controller 节点 -->
    <node pkg="test_controller" type="controller" name="controller" output="screen">
        <param name="controller_param" value="some_value" />
        <param name="simulation" value="$(arg simulation)" />
        <param name="uav_id" value="$(arg uav_id)" />
        <param name="file_path" value="$(arg file_path)" />
        <param name="controller_rate" value="$(arg controller_rate)" />

    </node>
    <!-- 启动 Observer 节点 -->
    <node pkg="test_controller" type="observer" name="observer" output="screen">
        <param name="observer_param" value="some_value" />
        <param name="uav_id" value="$(arg uav_id)" />
        <param name="simulation" value="$(arg simulation)" />
        <param name="file_path" value="$(arg file_path)" />
        <param name="observer_rate" value="$(arg observer_rate)" />
    </node>
    <!-- 启动 Estimator 节点 -->
    <!-- <node pkg="test_controller" type="estimator" name="estimator" output="log">
        <param name="estimator_param" value="some_value" />
        <param name="uav_id" value="$(arg uav_id)" />
    </node> -->



</launch>
